mongorestore/restore.go

```diff
@@ -340,6 +343,26 @@ func convertLegacyIndexes(indexes []IndexDocument) {
 	}
 }
 
+func fixDottedHashedIndexes(indexes []IndexDocument) {
+	for _, index := range indexes {
+		fixDottedHashedIndex(index)
+	}
+}
+
+// fixDottedHashedIndex fixes the issue introduced by a server bug where hashed index constraints are not
+// correctly enforced under all circumstance by changing the hashed index on the dotted field to an
+// ascending single field index.
+func fixDottedHashedIndex(index IndexDocument) {
```

Comment:
At the moment, this is duplicated between here and mongomirror. It would be nice if they shared the code -- could we either do that now, or file a ticket to do so in the future? (I don't really have a preference one way or the other)

Reply:
They aren't, though. I thought the same, but mongo mirror is operating on bson.D, and this works on IndexDocuments. 

I think filing a ticket would be good, in which case we can change MongoMirror to use IndexDocument instead (I think).

Reply:
Filing a ticket for MONGOMIRROR